client/www/components/dash/Sandbox.tsx

```diff
@@ -1,31 +1,71 @@
-import { init as initAdmin, tx, id, lookup } from '@instantdb/admin';
+import { id, init as initAdmin, lookup, tx } from '@instantdb/admin';
 import { init as initCore, InstantUnknownSchema } from '@instantdb/core';
 import Json from '@uiw/react-json-view';
-import { lightTheme } from '@uiw/react-json-view/light';
 import { darkTheme } from '@uiw/react-json-view/dark';
+import { lightTheme } from '@uiw/react-json-view/light';
 
-import config, { getLocal, setLocal } from '@/lib/config';
-import { InstantApp } from '@/lib/types';
-import { Editor } from '@monaco-editor/react';
-import { useEffect, useRef, useState } from 'react';
-import { Button, Checkbox, Label, TextInput } from '@/components/ui';
-import { dbAttrsToExplorerSchema } from '@/lib/schema';
-import clsx from 'clsx';
+import {
+  Button,
+  Checkbox,
+  cn,
+  Dialog,
+  IconButton,
+  Label,
+  Select,
+  TextInput,
+  useDialog,
+} from '@/components/ui';
+import config from '@/lib/config';
 import useLocalStorage from '@/lib/hooks/useLocalStorage';
+import { dbAttrsToExplorerSchema } from '@/lib/schema';
+import { DBAttr, InstantApp } from '@/lib/types';
 import {
   Combobox,
   ComboboxInput,
   ComboboxOption,
   ComboboxOptions,
 } from '@headlessui/react';
+import {
+  ArchiveBoxArrowDownIcon,
+  ArrowPathIcon,
+  PlayIcon,
+  PlusIcon,
+  TrashIcon,
+} from '@heroicons/react/24/outline';
 import { InstantReactWebDatabase } from '@instantdb/react';
+import { Editor } from '@monaco-editor/react';
+import clsx from 'clsx';
+import { createParser, parseAsBoolean, useQueryState } from 'nuqs';
+import { useEffect, useRef, useState } from 'react';
+import {
+  ResizableHandle,
+  ResizablePanel,
+  ResizablePanelGroup,
+} from '../resizable';
 import { useDarkMode } from './DarkModeToggle';
+import { ArrowUpToLine, Save } from 'lucide-react';
 
-let cachedSandboxValue = '';
+const base64Parser = createParser({
+  parse(value) {
+    try {
+      return JSON.parse(atob(value));
+    } catch {
+      return '';
+    }
+  },
+  serialize(value) {
+    return btoa(JSON.stringify(value));
```

Comment:
You might want to make this url-safe, `btoa(JSON.stringify(value)).replace(/+/g, '-').replace(/\//g, '_').replace(/=/g, '');` here and `JSON.parse(atob(value).replace(/-/g, '+').replace(/_/g, '/'));` above.